transport/src/main/java/io/netty/channel/ChannelConfig.java

```diff
@@ -253,4 +253,12 @@
      * to detect the size of a message.
      */
     ChannelConfig setMessageSizeEstimator(MessageSizeEstimator estimator);
+
+    /**
+     * Set the {@link WriteBufferWaterMark} which is used for set the high and low
+     * water mark of the write buffer.
+     */
+    ChannelConfig setWriteBufferWaterMark(WriteBufferWaterMark writeBufferWaterMark);
```

Comment:
maybe we could `@deprecated` the old methods and `ChannelConfig` ? I think users should always us the new one.